### PR TITLE
Add a GradleUp Governance Charter

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change introduces the initial Governance Charter, including the positioning of the organization, legal notice, decision making process and licenses. I built it based on the Jenkins and WireMock governance models. I tried to keep it as lightweight as possible

### CODE REVIEW

File: docs/community/governance.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+We host components and ensure their long-term sustainability.
+
+Our scope includes all things that help to achieve great user and developer experience when working with Gradle.
+It includes but not limited to plugins, developer tools, integrations, documentation or community content.
+
+## Formal Side
+
+The GradleUp organization is a volunteer driven community
+operating in a public interest in a non-profit mode and
+providing no commercial services.
```
Comment (martinbonnin) [inline docs/community/governance.md:17]: Should we explicit that there is no legal structure for GradleUp?

File: docs/community/governance.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+As it is written in the Apache License v2:
+
+> Licensor provides the Work (and each Contributor provides its Contributions) on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied, including, without limitation, any warranties or conditions of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A PARTICULAR PURPOSE. You are solely responsible for determining the appropriateness of using or redistributing the Work and assume any risks associated with Your exercise of permissions under this License.
+
+Gradle and the Gradlephant logo (the "Gradle Marks") are registered trademarks of [Gradle, Inc.](https://gradle.com/) and/or its subsidiaries.
+In the GradleUp organization, "Gradle" typically means "Gradle Build Tool" and does not reference Gradle Inc. and/or its subsidiaries.
+All references to the trademark represent a fair use,
+and do not imply endorsement.
```
Comment (martinbonnin) [inline docs/community/governance.md:26]: 👍

File: docs/community/governance.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+## Structure
+
+The project is led by a small team of passionate Gradle users and contributors.
+This [GradleUp Team](./team.md) is to be considered as a top governance and steering entity for the project.
+
+The project also includes maintainers and contributors.
```
Comment (martinbonnin) [inline docs/community/governance.md:33]: I'd remove that section and focus on roles below

File: docs/community/governance.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+* **Team Member** -
+  A GradleUp contributor listed on the [GradleUp Team page](./team.md).
+  They participate in the project steering as whole.
+  New team members are added by the decision of the current team members.
+* **Org Admin** - A team member with administrative access to the GitHub
+  organization and/or other community resources.
+  Permissions and access are granted based on the decision of the current team members.
+* **Maintainer** - A GradleUp contributor who maintains a component hosted on GradleUp.
+  It includes roles _Triage_ or above.
+  Components may explicitly or implicitly introduce more Maintainer sub-roles.
```
Comment (martinbonnin) [inline docs/community/governance.md]: Suggestion to simplify and have only 3 roles:

- Member
  - access to a specific GitHub repository
- Maintainer
  - admin access to a specific GitHub repository
- Admin
  - admin access to GitHub org, Sonatype, Gradle plugin portal and DNS

Contributors can be anyone, right, we do accept pull requests, etc... Just like any other OSS orgs. I don't expect them to be part of the GradleUp team (unless they want to of course)

File: docs/community/governance.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  
+## Decision Making
+
+The decisions are made by the [GradleUp Team](./team.md) based on consensus and common sense.
+We intend to make decision making fast and light-weight,
+with minimum overhead when a consensus exists.
+
+At the moment the decisions are made in a GradleUp Team private channel.
+Once [public communication channels](./participate.md) are established,
+it is expected that most of the discussions will happen there,
```
Comment (martinbonnin) [inline docs/community/governance.md:60]: Agreed

File: docs/community/governance.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+and all contributors and end users will be able to contribute to the decisions.
+
+## Components Governance
+
+### Component Ownership
+
+_Maintainers_ have full control of technical decisions and processes within their components.
+Apart from the requirements documented in the [Hosting Guidelines](./hosting.md),
```
Comment (martinbonnin) [inline docs/community/governance.md:68]: 👍 We should add a note in `hosting.md` that maven coordinates must be in the `com.gradleup` namespace

File: docs/community/governance.md (hunk 1 above)
Comment (oleg-nenashev) [reply docs/community/governance.md:17]: Agreed

```suggestion
The GradleUp organization is a volunteer driven community
operating in a public interest in a non-profit mode and
providing no commercial services.
There is no formal legal entity behind GradleUp.
```

File: docs/community/governance.md (hunk 6 above)
Comment (oleg-nenashev) [reply docs/community/governance.md:68]: I prefer to postpone it until there is a trademark usage permission

File: docs/community/governance.md (hunk 4 above)
Comment (oleg-nenashev) [reply docs/community/governance.md]: It excludes the "GradleUp Team member" which is the current steering entity. I would prefer to keep it as is for now, given we may want to have less org admins with permissions.

File: docs/community/governance.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+The project is led by a small team of passionate Gradle users and contributors.
+This [GradleUp Team](./team.md) is to be considered as a top governance and steering entity for the project.
+
+The project also includes maintainers and contributors.
+
+## Roles
+
+* **Contributor** - any individual who participates in the community and 
+  adds value to in one form or another,
+  including but not limited to code, documentation ,automation, advocacy or
```
Comment (leonard84) [inline docs/community/governance.md]:
```suggestion
  including but not limited to code, documentation, automation, advocacy or
```

File: docs/community/governance.md (hunk 4 above)
Comment (martinbonnin) [reply docs/community/governance.md]: > It excludes the "GradleUp Team member" which is the current steering entity

I'd say everyone in the GradleUp organization would effectively be a team member (https://github.com/orgs/GradleUp/people)? 

* You're a repo maintainer => you're part of the GradleUp org
* You're a repo maintainer and you invite a new collaborator => they're part of the GradleUp org

There might be other ways but it's handy to have https://github.com/orgs/GradleUp/people as the source of truth for the "GradleUp" team.

File: docs/community/team.md [hunk 8]
```diff
@@ -15,4 +16,5 @@ The team is decentralized and decisions are made based on consensus and common s
 
 ## Contact the Team
 
-For now, use DMs on the Kotlin Foundation or the Gradle Community Slack.
+For now, we use DMs on the Kotlin Foundation or the Gradle Community Slack.
```
Comment (LouisCAD) [inline docs/community/team.md]:
```suggestion
For now, we use DMs on the Kotlinlang Slack, or the Gradle Community Slack.
```

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**